packages/ui/src/lib/tooltip/Tooltip.spec.ts

```diff
@@ -40,10 +70,49 @@ test('tooltip is not empty string when tooltipHidden value false', async () => {
   expect(screen.queryByText('test 1')).toBeInTheDocument();
 });
 
-test('tooltip z order', async () => {
+test('Expect tooltip z order', async () => {
   render(Tooltip, { tip: 'my tooltip' });
 
   // get the tooltip
   const tooltip = screen.getByText('my tooltip');
   expect(tooltip.parentElement).toHaveClass('z-60');
 });
+
+test('Expect class styling to apply to tip', async () => {
+  render(Tooltip, { class: 'my-[5px] mx-[10px]', tip });
+
+  const slotElement = screen.getByLabelText('tooltip');
+  expect(slotElement).toHaveClass('my-[5px] mx-[10px]');
+});
+
+test('Expect class styling to apply to tip snippet', async () => {
+  render(Tooltip, {
+    class: 'my-[5px] mx-[10px]',
+    tip: createRawSnippet(() => {
+      return {
+        render: (): string => 'test',
+      };
+    }),
+  });
+
+  const slotElement = screen.getByLabelText('tooltip');
+  expect(slotElement).toHaveClass('my-[5px] mx-[10px]');
+});
+
+function createTest(props: Record<string, boolean>, locationName: string, expectedStyle = locationName): void {
+  test(`Expect property ${locationName} to add ${expectedStyle} class to parent element`, () => {
+    render(Tooltip, { tip, ...props });
+    const element = screen.getByLabelText('tooltip');
+    expect(element).toBeInTheDocument();
+    expect(element.parentElement).toHaveClass(expectedStyle);
+  });
+}
+
+createTest({ left: true }, 'left');
+createTest({ right: true }, 'right');
+createTest({ bottom: true }, 'bottom');
+createTest({ top: true }, 'top');
+createTest({ topLeft: true }, 'topLeft', 'top-left');
+createTest({ topRight: true }, 'topRight', 'top-right');
+createTest({ bottomLeft: true }, 'bottomLeft', 'bottom-left');
+createTest({ bottomRight: true }, 'bottomRight', 'bottom-right');
```

Comment:
does not seems to match our current testing practice 

You can take a look at test.each documentation https://vitest.dev/api/#test-each

Reply:
Tests were unchanged, just moved from other file. Updated to use test.each via new commit.